test/chplcheck/CaseRules.chpl

```diff
@@ -0,0 +1,41 @@
+module CaseRules {
+    var camelVar: int;
+    const PascalVar: real;
+    var snake_var: string;
+
+    record camelRecord {
+        proc camelCaseMethod() {return 0;}
+        proc PascalCaseMethod() {return 0;}
+        proc snake_case_method() {return 0;}
+    }
+    record PascalRecord {}
+    record snake_record {
+        var camelField, anotherCamelField: int;
+        const camelField2, PascalField: real;
+        var camelField3, snake_field: string;
+    }
+
+    proc camelProc() {return 0;}
+    proc PascalProc() {return 0;}
+    proc snake_proc() {
```

Comment:
I can see this being problematic for `extern` procs. Could you add a test case for a `snake_case` extern function and modify the rule + this test so there's no warning?